### PR TITLE
Upgrade dependencies and update server-extension install instructions

### PR DESCRIPTION
**Pull request by hbcarlos**
I used the script to upgrade all extensions at once and check that every extension works.

I also modified the installation instructions of the server-extension to to add the enable server extension command.

### CODE REVIEW

**Comment by fcollonval** [comment]
We need to figure out why eslint is broken :'(

**Comment by hbcarlos** [comment]
Could it be that it is linting files in `node_modules`?

**Comment by fcollonval** [comment]
> Could it be that it is linting files in `node_modules`?

I don't think so. This project is not freezing its `yarn.lock` file and `eslint` is working on my local copy. But I presume that if the node modules and the `yarn.lock` files were removed, it will raise the same error. This sounds like new eslint dependencies were released that bring issues.

**Comment by hbcarlos** [comment]
Yeah, I added the `package-lock.json` and it has `eslint v7.31`.

Does the `jupyter-packaging` uses npm if yarn is not installed?

**Comment by fcollonval** [comment]
> Does the `jupyter-packaging` uses npm if yarn is not installed? Or if you don't indicate the build tool?
> https://github.com/hbcarlos/extension-examples/blob/006053e190f737c911437ab6654bc0ac386cc777/hello-world/setup.py#L70

Yes it does: https://github.com/jupyter/jupyter-packaging/blob/2e22a16559601a6830a2d5bb263be4ccd1a48690/jupyter_packaging/setupbase.py#L189

**Comment by fcollonval** [comment]
There has been lots of new release around jsdoc lately. Could try updating `eslint-plugin-jsdoc` to `^36.0.0` in all examples? Hopefully this will fix the problem.

**Comment by hbcarlos** [comment]
Thanks for these tips, @fcollonval.

I think the test failures are coming from the UI tests because of the timeout but I am not sure what to do.

**Comment by fcollonval** [comment]
> I think the test failures are coming from the UI tests because of the timeout but I am not sure what to do.

The error is probably coming because we are downloading the latest prerelease version of JupyterLab from pypi.org (I needed 3.1). Now that 3.1 has been released, I remove the `--pre` option because it was downloading the 4.0 alpha in which the view menu has changed. That changes make closing the file browser code in the test not working any more.

**Review by fcollonval — APPROVED** [review]
Thanks a lot @hbcarlos :tada:

File: server-extension/README.md [hunk 1]
```diff
@@ -660,7 +670,7 @@ JupyterLab is looking for frontend extensions when the Python package is install
 ```py
 # setup.py#L25-L25
 
-("share/jupyter/labextensions/%s" % labext_name, str(lab_path.relative_to(HERE)), "**"),
+data_files_spec = [
```
Comment (fcollonval) [inline server-extension/README.md:673]: Line reference needs to be updated

File: server-extension/README.md [hunk 2]
```diff
@@ -686,7 +696,7 @@ in the appropriate jupyter folder (`etc/jupyter/jupyter_server_config.d`):
 ```py
 # setup.py#L27-L27
 
-("etc/jupyter/jupyter_server_config.d", "jupyter-config/server-config", "jlab_ext_example.json"),
+("share/jupyter/labextensions/%s" % labext_name, str("."), "install.json"),
```
Comment (fcollonval) [inline server-extension/README.md:699]: Line reference needs to be updated

File: server-extension/README.md [hunk 3]
```diff
@@ -695,7 +705,7 @@ For backward compatibility with the classical notebook, the old version of that
 ```py
 # setup.py#L29-L29
 
-("etc/jupyter/jupyter_notebook_config.d", "jupyter-config/nb-config", "jlab_ext_example.json"),
+"jupyter-config/server-config", "jlab_ext_example.json"),
```
Comment (fcollonval) [inline server-extension/README.md:708]: Line reference needs to be updated

**Review by fcollonval — CHANGES_REQUESTED** [review]
Sorry I miss some mistake in the embed snippets

File: server-extension/README.md [hunk 4]
```diff
@@ -761,6 +771,8 @@ to get you set are:
 pip install -e .
 # Link your development version of the extension with JupyterLab
 jupyter labextension develop . --overwrite
+# Enable the server extension
+jupyter server extension enable jlab_ext_example
```
Comment (fcollonval) [inline server-extension/README.md:775]: Thanks for correcting that one

File: server-extension/README.md [hunk 5]
```diff
@@ -668,9 +668,9 @@ It will copy the NPM package in the Python package and force it to be copied in
 JupyterLab is looking for frontend extensions when the Python package is installed:
 
 ```py
-# setup.py#L25-L25
+# setup.py#L26-L26
 
-data_files_spec = [
+("share/jupyter/labextensions/%s" % labext_name, str(lab_path.relative_to(HERE)), "**"),
```
Comment (hbcarlos) [inline server-extension/README.md]: Did you mean this?

**Review by fcollonval — APPROVED** [review]
> Did you mean this?

Yes thanks for correcting them.

**Pull request merged**